mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,815 @@
+        ...
+
+    Example of a standard annotation file:
+    .. code-block:: txt
+        # 0
+        video_validation_0000202
+        5666
+        1
+        3
+        8 130 185
+        8 832 1136
+        8 1303 1381
+        5
+        8 0.0620 0.0620 790 5671
+        8 0.1656 0.1656 790 2619
+        8 0.0833 0.0833 3945 5671
+        8 0.0960 0.0960 4173 5671
+        8 0.0614 0.0614 3327 5671
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        train_cfg (dict): Config for training.
+        test_cfg (dict): Config for testing.
+        data_prefix (str): Path to a directory where videos are held.
+        test_mode (bool): Store True when building test or validation dataset.
+            Default: False.
+        filename_tmpl (str): Template for each filename.
+            Default: 'img_{:05}.jpg'.
+        start_index (int): Specify a start index for frames in consideration of
+            different filename format. Default: 1.
+        modality (str): Modality of data. Support 'RGB', 'Flow'.
+            Default: 'RGB'.
+        video_centric (bool): Whether to sample proposals just from
+            this video or sample proposals randomly from the entire dataset.
+            Default: True.
+        reg_normalize_constants (list): Regression target normalized constants,
+            including mean and standard deviation of location and duration.
+        body_segments (int): Number of segments in course period.
+            Default: 5.
+        aug_segments (list[int]): Number of segments in starting and
+            ending period. Default: (2, 2).
+        aug_ratio (int | float | tuple[int | float]): The ratio of the length
+            of augmentation to that of the proposal. Defualt: (0.5, 0.5).
+        clip_len (int): Frames of each sampled output clip.
+            Default: 1.
+        frame_interval (int): Temporal interval of adjacent sampled frames.
+            Default: 1.
+        filter_gt (bool): Whether to filter videos with no annotation
+            during training. Default: True.
+        no_regression (bool): Whether to perform regression. Default: False.
+        verbose (bool): Whether to print full information or not.
+            Default: False.
+    """
+
+    def __init__(self,
+                 ann_file,
+                 pipeline,
+                 train_cfg,
+                 test_cfg,
+                 data_prefix,
+                 test_mode=False,
+                 filename_tmpl='img_{:05d}.jpg',
+                 start_index=1,
+                 modality='RGB',
+                 video_centric=True,
+                 reg_normalize_constants=None,
+                 body_segments=5,
+                 aug_segments=(2, 2),
+                 aug_ratio=(0.5, 0.5),
+                 clip_len=1,
+                 frame_interval=1,
+                 filter_gt=True,
+                 no_regression=False,
```

Comment:
Should we use `has_regression` for consistance ?

Reply:
Agree.